Mathlib/RingTheory/PiTensorProduct.lean

```diff
@@ -0,0 +1,270 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import Mathlib.LinearAlgebra.PiTensorProduct
+import Mathlib.Algebra.Algebra.Bilinear
+
+/-!
+# Tensor product of `R`-algebras and rings
+
+If `(Aᵢ)` is a family of `R`-algebras then the `R`-tensor `⨂ᵢ Aᵢ` is an `R`-algebra as well with
+its structure map defined by `r ↦ r • 1`.
+
+In particular if we take `R` to be `ℤ`, then this collapse into tensor product of rings.
+-/
+
+open TensorProduct Function
+
+variable {ι R : Type*} {A : ι → Type*}
+
+namespace PiTensorProduct
+
+noncomputable section AddCommMonoidWithOne
+
+variable [CommSemiring R] [∀ i, AddCommMonoidWithOne (A i)] [∀ i, Module R (A i)]
+
+instance one : One (⨂[R] i, A i) where
+  one := tprod R 1
+
+instance addCommMonoidWithOne : AddCommMonoidWithOne (⨂[R] i, A i) where
+  __ := inferInstanceAs (AddCommMonoid (⨂[R] i, A i))
```

Comment:
@eric-wieser is using `__` OK here? I mean, should we worry about how these instances are defined or now everything should just work?

Reply:
Everything should just work I think. Certainly `__` is no worse than `with`